modules/core/src/v2/coins/eos.ts

```diff
@@ -421,25 +468,80 @@ export class Eos extends BaseCoin {
       }
-      if (txAction.name !== 'transfer') {
+
+      if (txAction.name === 'transfer') {
+        // Transfers should only have 1 action
+        if (tx.actions.length !== 1) {
+          throw new Error(`transfers should only have 1 action: ${tx.actions.length} given`);
+        }
+
+        const transferStruct = eosClient.fc.abiCache.abi('eosio.token').structs.transfer;
+        const serializedTransferDataBuffer = Buffer.from(txAction.data, 'hex');
+        const transferActionData = EosJs.modules.Fcbuffer.fromBuffer(transferStruct, serializedTransferDataBuffer);
+        tx.address = transferActionData.to;
+        tx.amount = this.bigUnitsToBaseUnits(transferActionData.quantity.split(' ')[0]);
+        tx.memo = transferActionData.memo;
+      } else if (txAction.name === 'delegatebw') {
+        // The delegatebw action should only be part of voting transactions
+        if (tx.actions.length !== 2) {
+          throw new Error(
+            `staking transactions that include the delegatebw action should have 2 actions: ${tx.actions.length} given`
+          );
+        }
+
+        const txAction2 = tx.actions[1];
+        if (txAction2.name !== 'voteproducer') {
+          throw new Error(`invalid staking transaction action: ${txAction2.name}, expecting: voteproducer`);
+        }
+
+        const deserializedStakeAction = self.deserializeStakeAction(eosClient, txAction.data);
+        const deserializedVoteAction = Eos.deserializeVoteAction(eosClient, txAction2.data);
+        if (deserializedStakeAction.address !== deserializedVoteAction.address) {
+          throw new Error(
+            `staker (${deserializedStakeAction.address}) and voter (${deserializedVoteAction.address}) must be the same`
+          );
+        }
+
+        tx.amount = deserializedStakeAction.amount;
+        tx.proxy = deserializedVoteAction.proxy;
+        tx.producers = deserializedVoteAction.producers;
+      } else if (txAction.name === 'voteproducer') {
+        if (tx.actions.length > 2) {
+          throw new Error('voting transactions should not have more than 2 actions');
+        }
+
+        let deserializedStakeAction;
+        if (tx.actions.length === 2) {
+          const txAction2 = tx.actions[1];
+          if (txAction2.name !== 'delegatebw') {
+            throw new Error(`invalid staking transaction action: ${txAction2.name}, expecting: delegatebw`);
+          }
+
+          deserializedStakeAction = self.deserializeStakeAction(eosClient, txAction2.data);
+        }
+
+        const deserializedVoteAction = Eos.deserializeVoteAction(eosClient, txAction.data);
+        if (!!deserializedStakeAction && deserializedStakeAction.address !== deserializedVoteAction.address) {
+          throw new Error(
+            `staker (${deserializedStakeAction.address}) and voter (${deserializedVoteAction.address}) must be the same`
+          );
+        }
+
+        tx.amount = deserializedStakeAction.amount;
```

Comment:
Potential issue: If `tx.actions.length` is 1 and `txAction.name` is 'voteproducer', won't this access a property of undefined?

Reply:
Great catch! Thank you. Fixed.